src/ingameWindows/iwMapGenerator.cpp

```diff
@@ -0,0 +1,313 @@
+// Copyright (c) 2005-2010 Settlers Freaks (sf-team at siedler25.org)
+//
+// This file is part of Siedler II.5 RTTR.
+//
+// Siedler II.5 RTTR is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 2 of the License, or
+// (at your option) any later version.
+//
+// Siedler II.5 RTTR is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Siedler II.5 RTTR. If not, see <http://www.gnu.org/licenses/>.
+
+#include "defines.h" // IWYU pragma: keep
+#include "iwMapGenerator.h"
+
+#include "Loader.h"
+#include "controls/ctrlOptionGroup.h"
+#include "controls/ctrlComboBox.h"
+#include "controls/ctrlProgress.h"
+
+#include "gameData/const_gui_ids.h"
+#include "helpers/containerUtils.h"
+#include "libutil/src/colors.h"
+
+#include <string>
+
+iwMapGenerator::iwMapGenerator(MapSettings& settings) : IngameWindow(CGI_MAP_GENERATOR,
+                                                IngameWindow::posLastOrCenter,
+                                                250, 400, _("Map Generator"),
+                                                LOADER.GetImageN("resource", 41), true, false), mapSettings(settings), tmpSettings(settings)
+{
+    AddTextButton(0,  20, 360, 100, 20, TC_RED2, _("Back"), NormalFont);
+    AddTextButton(1, 130, 360, 100, 20, TC_GREEN2, _("Apply"), NormalFont);
+
+    ctrlComboBox* combo = AddComboBox(2, 20, 30, 210, 20, TC_GREY, NormalFont, 100);
+    combo->AddString("2 Players");
```

Comment:
This can be a loop and also internationalized. Look for `_(...)` and boost::format (already used)